app/src/main/java/openfoodfacts/github/scrachx/openfood/views/LoginActivity.java

```diff
@@ -88,21 +88,21 @@ protected void onCreate(Bundle savedInstanceState) {
 
         setTitle(getString(R.string.txtSignIn));
         setSupportActionBar(toolbar);
-        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
-
+        if(getSupportActionBar()!=null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        }
         userLoginUri = Uri.parse(getString(R.string.website) + "cgi/user.pl");
         resetPasswordUri = Uri.parse(getString(R.string.website) + "cgi/reset_password.pl");
 
         // prefetch the uri
         customTabActivityHelper = new CustomTabActivityHelper();
         customTabActivityHelper.setConnectionCallback(this);
         customTabActivityHelper.mayLaunchUrl(userLoginUri, null, null);
-
-        signup.setEnabled(false);
+        signup.setEnabled(true);
 
         final SharedPreferences settings = getSharedPreferences("login", 0);
         String loginS = settings.getString("user", getResources().getString(R.string.txt_anonymous));
-        if (!loginS.equals(getResources().getString(R.string.txt_anonymous))) {
+        if (loginS.equals("user")) {
```

Comment:
Please use a string resource here rather than the hardcoded string "user". This allows the app to be multilingual.